src/lib.rs

```diff
@@ -347,6 +355,43 @@ impl Rng {
         f64::from_bits((1 << (b - 2)) - (1 << f) + (self.u64(..) >> (b - f))) - 1.0
     }
 
+    /// Collects `amount` values at random from the iterator into a vector.
+    ///
+    /// The length of the returned vector equals `amount` unless the iterator
+    /// contains insufficient elements, in which case it equals the number of
+    /// elements available.
+    ///
+    /// Complexity is `O(n)` where `n` is the length of the iterator.
+    pub fn choose_multiple<T: std::iter::Iterator>(
+        &self,
+        mut source: T,
+        amount: usize,
+    ) -> Vec<T::Item> {
```

Comment:
If this function returns a `Vec`, then it should only be enabled on platforms with an allocator. Could you add an `alloc` feature for this?

Reply:
ah yeah I think this PR was made before the crate was made `no_std`...